### DISCUSSION TITLE
What kind of LCD protocol is this?

### DISCUSSION THREAD

**Discussion by simongrabar** [olikraus/u8g2#1418] [Q&A]
Hi all,

We are trying to reuse an existing 3,5" monochrome LCDs, and drive them with new microcontroller, but we are unable to determine what protocol LCD is using. Attached are traces from Analyser. Short Spikes on D12,D13 are probably only noise since LCD works OK without those lines being connected. D4-D11 are 8 data lines. First Pic is complete frame, other two are zoom-In, with last one showing start sequence of each frame.

![CompleteFrame](https://user-images.githubusercontent.com/16597849/109384287-18493d00-78ec-11eb-90da-f6557b0dc8b2.png)
![StartFrameSequenceZoomOut](https://user-images.githubusercontent.com/16597849/109384288-18e1d380-78ec-11eb-90ad-34fe266017c0.png)
![StartFrameSequence](https://user-images.githubusercontent.com/16597849/109384289-18e1d380-78ec-11eb-8464-a53a9ba8f99c.png)

**Comment by olikraus** [comment]
Difficult. Any labels on the display itself?

**Reply by simongrabar** [reply]
Its OEM made automotive display LAM0353629B probably from Japan Display, so no way to get a datasheet directly from JD. 
There are only 3 control and 8 data bytes, that are changing. Other lines are constant 3.3VCC or GND.
It looks like there is some sort of write registers at beginning of each frame and then followed by data. 

![WhatsApp Image 2020-09-04 at 11 07 50 (2) (1)](https://user-images.githubusercontent.com/16597849/109386342-551c3080-78fa-11eb-98bd-c4cf2f5efa91.jpeg)
![2020-12-2 14-26-57](https://user-images.githubusercontent.com/16597849/109386439-e4294880-78fa-11eb-84fc-da23801659ca.jpg)
![2020-12-2 14-27-20](https://user-images.githubusercontent.com/16597849/109386440-e4c1df00-78fa-11eb-8ec7-46335d4fc305.jpg)

**Comment by olikraus** [comment]
Are you able to extract first 20 Byte values which are sent do the display?
Another question: what is the resolution of the display?

**Reply by simongrabar** [reply]
What do you mean by first bytes? First bytes after power up, or at beginning of each frame, like seen on 3rd pic? For resolution I need to count pixels, will do it asap.

**Reply by olikraus** [reply]
Yes, the first bytes of the init sequence.

**Reply by simongrabar** [reply]
Measured pixel size is 0,224 mm and usable size 68,6(it can be a bit more, but I cant illuminate them)x53,6 so measured resolution is 311x243, which would make it 340x240. 
First bytes of sequence are seen on picture I already attached, but its hard to say what are bits since I don't know the protocol. D1 looks like RS by 8080 protocol triggered on falling edge?
![image](https://user-images.githubusercontent.com/16597849/109490759-f76a1e80-7a88-11eb-9b0b-26a7c9127691.png)

**Reply by simongrabar** [reply]
It looks so simple, just 3 control lines and data, but we are not able to make it alive.

**Reply by simongrabar** [reply]
Timing of first 3 pictures explained
![image](https://user-images.githubusercontent.com/16597849/109492026-a0654900-7a8a-11eb-85fd-9f13420e3b3c.png)

**Reply by olikraus** [reply]
It looks like a COG display.The only COG controller with the mentioned dimension, which is supported by u8g2 is the ST75320 controller.

**Comment by simongrabar** [comment]
Is anyone here interested to develop a driver for this LCD?

**Reply by olikraus** [reply]
I still feel that this is difficult. The ST75320 is a wild guess. If signal D3 is the A0 signal of the datasheet, then this does not fit at all (there is no display command with four arguments). However the first byte could be the power off command (which is suggested to be the first command according to the datasheet). In fact also the second byte (first arg to the power off) would fit then. But third byte would not fit.
